src/name_bazaar/server/contracts_api/registrar.cljs

```diff
@@ -17,19 +16,22 @@
                          :value (web3/to-wei 0.01 :ether)}
                         opts)))
 
-(defn transfer! [{:keys [:ens.record/hash :ens.record/label :ens.record/owner]} & [opts]]
-  (contract-call :name-bazaar-registrar
-                 :transfer
-                 (sha3 label hash)
-                 owner
-                 (merge {:gas 2000000}
-                        opts)))
-
-(defn entry [{:keys [:ens.record/hash :ens.record/label]}]
-  (parse-registrar-entry (contract-call :name-bazaar-registrar :entries (sha3 label hash))))
+(defn transfer! [{:keys [:ens.record/label :ens.record/owner]} & [opts]]
+  {:set-owner-tx
+     (ens/set-owner! {:ens.record/name (str label ".eth")
+                      :ens.record/owner owner}
+                     opts)
+   :transfer-tx
+     (contract-call :name-bazaar-registrar
+                    :transferFrom
+                    (:from opts)
+                    owner
+                    (sha3 label hash)
+                    (merge {:gas 2000000}
+                           opts))})
 
-(defn entry-deed-owner [args & [opts]]
-  (deed/owner (:name-bazaar-registrar.entry.deed/address (entry args))))
+(defn registration-owner [{:keys [:ens.record/hash :ens.record/label]}]
```

Comment:
I understand roughly what this does (destructuring), but do you have a source where I can read more about this (what's supported and what's the syntax for it)?

Reply:
https://clojure.org/guides/destructuring